### ISSUE TITLE
x/tools/gopls: bad completion after dangling selector in switch statement

### ISSUE THREAD

**Issue by muirdm** [golang/go#44443]
In the below example:

```go
var s struct{ foo int }
switch {
case true:
	s.<>
default:
}
```

We don't complete properly to "foo" at `<>`. Our dangling selector fix does not work in this case for some reason.

**Comment by findleyr** [comment]
Which Go version are you using? This is likely to be a parser issue, and I'm planning to address this class of problem soon, having [just been bitten by it](https://golang.org/cl/293792).

**Comment by muirdm** [comment]
This was with Go 1.15.6. Looking forward to you fixing everything.

**Comment by stamblerre** [comment]
I can confirm that this is fixed with Go 1.17.
